### PR TITLE
MeleeHitSoundSystem

### PR DESCRIPTION
**Pull request by Bixkitts**
## About the PR

Separated melee sound stuff into a system

## Why / Balance

Clean the code, learn the code, and get it ready for more changes I want to do.
My ultimate plan is to make MeleeSoundComponents placed on clothing transfer their hitsounds to
the players wearing them too.

## Technical details

There was already a MeleeSoundComponent with non-trivial behaviour pertaining to it.
I've moved that to it's own system.

## Media

- [ X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Bixkitts** [comment]
As mentioned in Technical Details, I want to have clothes with a MeleeSoundComponent play their sound
when a player wearing them is hit.
I've considered overriding/adding to the MeleeSoundComponent dictionaries on the player OnEquip?
Do it for specific clothing with custom behaviour?
Do events propagate to worn clothing?
idek!

File: Content.Shared/Weapons/Melee/MeleeSoundSystem.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+using Robust.Shared.Audio.Systems;
+
+namespace Content.Shared.Weapons.Melee.Components;
+
+/// <summary>
+/// This handles <see cref="MeleeSoundComponent"/>
+/// </summary>
+public sealed class MeleeSoundSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem Audio = default!;
```
Comment (TheShuEd) [inline Content.Shared/Weapons/Melee/MeleeSoundSystem.cs:11]: _audio

**Review by TheShuEd — COMMENTED** [review]
a few stylistic quibbles

File: Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs [hunk 2]
```diff
@@ -36,22 +34,21 @@ namespace Content.Shared.Weapons.Melee;
...
-    [Dependency] protected readonly SharedAudioSystem Audio = default!;
-    [Dependency] protected readonly SharedCombatModeSystem CombatMode = default!;
-    [Dependency] protected readonly SharedInteractionSystem Interaction = default!;
-    [Dependency] private   readonly SharedPhysicsSystem _physics = default!;
-    [Dependency] protected readonly SharedPopupSystem PopupSystem = default!;
-    [Dependency] protected readonly SharedTransformSystem TransformSystem = default!;
-    [Dependency] private   readonly StaminaSystem _stamina = default!;
-
-    public const float DamagePitchVariation = 0.05f;
+    [Dependency] protected readonly IGameTiming              Timing          = default!;
```
[the remark this answers is not in the document]
Comment (TheShuEd) [reply Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs:37]: Oh ok

**Pull request merged**